src/snowflake/snowpark/modin/plugin/_internal/ordered_dataframe.py

```diff
@@ -1377,6 +1377,7 @@ def align(
             how: We support the following align/join types:
```

Comment:
Do we know why `right` is not one of the supported options here?

Reply:
I'm not sure why actually, perhaps there has not been a use case for it before. I rearranged the left align objects to implement right align